cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -217,13 +216,34 @@ func getAPIServerCommand(cfg *kubeadmapi.MasterConfiguration) []string {
 			defaultArguments["audit-log-maxage"] = fmt.Sprintf("%d", *cfg.AuditPolicyConfiguration.LogMaxAge)
 		}
 	}
-
+	if cfg.APIServerExtraArgs == nil {
+		cfg.APIServerExtraArgs = map[string]string{}
+	}
+	cfg.APIServerExtraArgs["authorization-mode"] = getAuthzModes(cfg)
 	command = append(command, kubeadmutil.BuildArgumentListFromMap(defaultArguments, cfg.APIServerExtraArgs)...)
-	command = append(command, getAuthzParameters(cfg.AuthorizationModes)...)
 
 	return command
 }
 
+// getAuthzModes gets the authorization-related parameters to the api server
+// Node,RBAC should be fixed in this order at the begining
+// AlwaysAllow and AlwaysDeny is ignored as they are only for testing
+func getAuthzModes(cfg *kubeadmapi.MasterConfiguration) string {
```

Comment:
I'd pass the string coming from `cfg.APIServerExtraArgs["authorization-mode"]` instead...
So the signature here should be `customAuthzModes string` instead. Then you don't need the `if cfg.APIServerExtraArgs != nil {` thing either.